Amplify/Categories/Auth/AuthCategory+UserBehavior.swift

```diff
@@ -13,6 +13,14 @@ extension AuthCategory: AuthCategoryUserBehavior {
         return plugin.getCurrentUser()
     }
 
+    public func getCurrentUser() async -> AuthUser? {
+        return await plugin.getCurrentUser()
+    }
```

Comment:
@royjit is it ok if we start providing async API's to customers?

Reply:
Since Auth already makes this process async we don't have a choice since we cannot block in this function.